source/slang/slang-ir-glsl-legalize.cpp

```diff
@@ -1337,153 +1384,375 @@ ScalarizedVal createSimpleGLSLGlobalVarying(
     OuterParamInfoLink* outerParamInfo,
     StringBuilder& nameHintSB)
 {
-    // Check if we have a system value on our hands.
+    // Note: The `leafUserDeclaredType` represents the type
+    // that was arrived at by recursively decomposing things
+    // like arrays and `struct` types. Any surrounding arrays
+    // (or array-like types such as mesh shader outputs) will
+    // be gathered up into the `declarator` that is passed down,
+    // so that we can reconstruct an SOA-ified version of the
+    // original parameter, for each leaf field/value.
+
+    // We start by checking if the context of this leaf field/value
+    // tells us that it should use a system-value semantic.
+    //
+    // Such a semantic should come from the `inVarLayout` that was
```

Comment:
Given the parameter rename, should the parameter be updated in the comment?
```suggestion
    // Such a semantic should come from the `userDeclaredParamVarLayout` that was
```